src/utils/progress.rs

```diff
@@ -42,6 +42,16 @@ impl ProgressBar {
     }
 
     pub fn finish_with_message(&self, msg: &str) {
+        self.inner.finish_with_message(msg.to_owned());
+        logging::set_progress_bar(None);
+    }
+
+    pub fn finish_with_duration(&self, op: &str) {
+        let e = self.inner.elapsed();
+        let msg = format!("{} completed in {}.{}s", op, e.as_secs(), e.as_millis());
```

Comment:
Do `as_secs()` and `as_millis()` give the second and millisecond parts of the time, or do they do what their names imply and give the same time expressed in different units? 

If the latter, it seems like this would be problematic in the case of something which took, say, 873 ms - in seconds that's 1 second, so you'd end up with `completed in 1.873s`, which obviously isn't correct.

Reply:
You are totally correct, great catch, thanks!